roles/rhel_subscribe/tasks/enterprise.yml

```diff
@@ -1,18 +1,47 @@
 ---
-- set_fact:
-    openshift_release: "{{ openshift_release[1:] }}"
-  when:
-  - openshift_release is defined
-  - openshift_release[0] == 'v'
-
-- name: Disable all repositories
-  command: subscription-manager repos --disable="*"
-
-- name: Enable RHEL repositories
-  command: subscription-manager repos \
-               --enable="rhel-7-server-rpms" \
-               --enable="rhel-7-server-extras-rpms" \
-               --enable="rhel-7-server-ose-{{ (openshift_release | default('')).split('.')[0:2] | join('.') }}-rpms" \
-               --enable="rhel-7-fast-datapath-rpms"
-  register: subscribe_repos
-  until: subscribe_repos | succeeded
+
+- name: Ensure RHEL rhui repositories are disabled
+  shell: yum -q --noplugins repolist | grep -v 'repo id' | grep rhui
+  register: repo_rhui
+  changed_when: "repo_rhui.rc != 1"
+  failed_when: repo_rhui.rc == 11
+
+- name: Disable RHEL rhui repositories
+  shell: yum-config-manager \
+              --disable "rhui-REGION-client-config-server-7" \
+              --disable "rhui-REGION-rhel-server-rh-common" \
+              --disable "rhui-REGION-rhel-server-releases" \
+              --disable "rhui-REGION-client-config-server-7"
+  when: repo_rhui.changed
+
+- name: Ensure RHEL repositories are enabled
+  shell: yum -q --noplugins repolist | grep -v 'repo id' | grep 'Red Hat' | wc -l
+  register: repo_rhel
+  changed_when: "'4' not in repo_rhel.stdout"
+  failed_when: repo_rhel.rc == 11
+
+- name: Enable correct RHEL repositories
+  block:
+    - name: Disable all repositories
+      shell: subscription-manager repos --disable='*'
+
+    - set_fact:
+        default_ose_version: '3.7'
+
+    - set_fact:
```

Comment:
We're really trying to reduce the use of set_fact, move this to defaults/main.yml ?

Reply:
Better yet... OpenShift version fail condition should not be here.  All the tasks relating to ose_version can simply be removed.